project/api.py

```diff
@@ -0,0 +1,204 @@
+
+    day_pivot = day_data.pivot(index='query_id', columns='status', values='time')
+    not_closed_within_day = day_pivot.isnull().values.any(1).nonzero()[0]  # if not empty returns indexes
+
+    if not_closed_within_day.shape[0] > 0:
+        day_pivot.drop(day_pivot.index[not_closed_within_day], inplace=True)
+
+    day_pivot['elapsed_time'] = (day_pivot[1] - day_pivot[0]) / 1000
+
+    avg_time = day_pivot['elapsed_time'].mean()
+
+    return jsonify({'Average elapsed time, sec': avg_time})
+
+
+@app.route('/api/v1/queries/avg_rows_per_sec', methods=['GET'])
+def avg_rows_per_sec():
+    """Returns an average per second number of rows returned from all queries during certain time range"""
+    query_parameters = request.args
+    time_range_start = query_parameters.get('start')
+    time_range_end = query_parameters.get('end')
+    if not time_range_start or not time_range_end:
+        return page_not_found('No time range provided')
+
+    try:
+        datetime_start_code, datetime_end_code = prepare_request(time_range_start, time_range_end)
+    except ValueError as e:
+        return handle_bad_request(e)
+
+    data = pd.read_csv('queries.tsv', sep='\t')
+
+    time_range_data = data[data['time'].between(datetime_start_code, datetime_end_code)]
+    if time_range_data.shape[0] == 0:
+        return handle_bad_request(400)
+
+    time_range_pivot = time_range_data.pivot(index='query_id', columns='status', values=['time', 'rows'])
+
+    not_closed_within_range = time_range_pivot.isnull().values.any(1).nonzero()[0]  # if not empty returns indexes
+
+    if not_closed_within_range.shape[0] > 0:
+        time_range_pivot.drop(time_range_pivot.index[not_closed_within_range], inplace=True)
+
+    time_range_pivot['elapsed_time'] = (time_range_pivot['time'][1] - time_range_pivot['time'][0]) / 1000
+
+    avg_rows = time_range_pivot['rows'][1].sum() / time_range_pivot['elapsed_time'].sum()
+    return jsonify({'Average rows per sec': avg_rows})
+
+
+@app.route('/api/v1/queries/avg_rows_per_thread', methods=['GET'])
+def avg_rows_per_thread():
+    """Returns an average per thread number of rows returned from all queries during certain time range"""
+    query_parameters = request.args
+    time_range_start = query_parameters.get('start')
+    time_range_end = query_parameters.get('end')
+    if not time_range_start or not time_range_end:
+        return page_not_found('No time range provided')
+
+    try:
+        datetime_start_code, datetime_end_code = prepare_request(time_range_start, time_range_end)
+    except ValueError as e:
+        return handle_bad_request(e)
+
+    data = pd.read_csv('queries.tsv', sep='\t')
+
+    time_range_data = data[data['time'].between(datetime_start_code, datetime_end_code) & data['status'] == 1]
+    if time_range_data.shape[0] == 0:
+        return handle_bad_request(400)
+
+    time_range_data['n_threads'] = time_range_data.threads.str.count(',') + 1
```

Comment:
Are there cases with zero threads in data? If there are such cases, then `n_threads` will be equal to 1, instead of 0.